src/test/java/org/minions/devfund/katerinaanzoleaga/WhackAMoleTest.java

```diff
@@ -0,0 +1,125 @@
+                {'*', '*', '*', '*'},
+                {'*', '*', '*', '*'}};
+        WhackAMole otherWhackAMole = new WhackAMole(NUM_ATEMPTS, GRID_DIMENSION);
+        otherWhackAMole.setMoleGrid(otherGrid);
+        assertTrue(whackAMole.equals(otherWhackAMole));
+    }
+
+    /**
+     * Verifies if it is possible to place a mole in a valid possiton.
+     */
+    @Test
+    public void placeWherePossible() {
+        char[][] otherGrid = new char[][]{
+                {'*', '*', '*', '*'},
+                {'*', '*', 'M', '*'},
+                {'*', '*', '*', '*'},
+                {'*', '*', '*', '*'}};
+        WhackAMole otherWhackAMole = new WhackAMole(NUM_ATEMPTS, GRID_DIMENSION);
+        otherWhackAMole.setMoleGrid(otherGrid);
+        assertTrue(whackAMole.place(1, 2));
+        assertTrue(whackAMole.equals(otherWhackAMole));
+    }
+
+    /**
+     * Verifies if it is NOT possible to place a mole in a NOT valid possiton.
+     */
+    @Test
+    public void placeWhereNotPossible() {
+        char[][] otherGrid = new char[][]{
+                {'*', '*', '*', '*'},
+                {'*', '*', '*', '*'},
+                {'*', '*', '*', '*'},
+                {'*', '*', '*', '*'}};
+        WhackAMole otherWhackAMole = new WhackAMole(NUM_ATEMPTS, GRID_DIMENSION);
+        otherWhackAMole.setMoleGrid(otherGrid);
+        assertFalse(whackAMole.place(1, OUT_OF_RANGE));
+        assertTrue(whackAMole.equals(otherWhackAMole));
+    }
+
+    /**
+     * Verifies the effect of placing a Mole in a valid position.
+     */
+    @Test
+    public void whackWerePosible() {
+        char[][] otherGrid = new char[][]{
+                {'*', '*', '*', '*'},
+                {'*', '*', '*', '*'},
+                {'*', '*', 'W', '*'},
+                {'*', '*', '*', '*'}};
+        WhackAMole otherWhackAMole = new WhackAMole(NUM_ATEMPTS, GRID_DIMENSION);
+        otherWhackAMole.setMoleGrid(otherGrid);
+        whackAMole.place(2, 2);
+        whackAMole.whack(2, 2);
+        assertEquals(NUM_ATEMPTS - 1, whackAMole.getAttemptsLeft());
+        assertEquals(0, whackAMole.getMolesLeft());
+        assertTrue(whackAMole.equals(otherWhackAMole));
+    }
+
+    /**
+     * Verifies the effect of TRYING TO place a Mole in a NOT valid position.
+     */
+
+    @Test
+    public void whackWereNotPosible() {
+        char[][] otherGrid = new char[][]{
+                {'*', '*', '*', '*'},
+                {'*', '*', '*', '*'},
+                {'*', '*', 'M', '*'},
+                {'*', '*', '*', '*'}};
+        WhackAMole otherWhackAMole = new WhackAMole(NUM_ATEMPTS, GRID_DIMENSION);
+        otherWhackAMole.setMoleGrid(otherGrid);
+        whackAMole.place(2, 2);
+        whackAMole.whack(OUT_OF_RANGE, 2);
+        assertEquals(NUM_ATEMPTS - 1, whackAMole.getAttemptsLeft());
+        assertEquals(1, whackAMole.getMolesLeft());
+        assertTrue(whackAMole.equals(otherWhackAMole));
+    }
+
+/*
+    @Test
+    public void printGridToUser() {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') This block of commented-out lines of code should be removed. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3ACommentedOutCodeLine)